src/components/NotificationsDrawer/DrawerPanelContent.tsx

```diff
@@ -39,19 +57,139 @@ const EmptyNotifications = () => (
 );
 
 const DrawerPanelBase = ({ innerRef }: DrawerPanelProps) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [isFilterDropdownOpen, setIsFilterDropdownOpen] = useState(false);
+  const [activeFilters, setActiveFilters] = useState<string[]>([]);
+  const [filteredNotifications, setFilteredNotifications] = useState<NotificationData[]>([]);
   const dispatch = useDispatch();
   const notifications = useSelector(({ chrome: { notifications } }: ReduxState) => notifications?.data || []);
+
+  useEffect(() => {
+    const modifiedNotifications = (activeFilters || []).reduce(
+      (acc: NotificationData[], chosenFilter: string) => [...acc, ...notifications.filter(({ source }) => source === chosenFilter)],
+      []
+    );
+
+    setFilteredNotifications(modifiedNotifications);
+  }, [activeFilters]);
+
+  const onNotificationsDrawerClose = () => {
+    setActiveFilters([]);
+    dispatch(toggleNotificationsDrawer());
+  };
+
+  const onMarkAllAsRead = () => {
+    dispatch(markAllNotificationsAsRead());
+    setIsDropdownOpen(false);
+  };
+
+  const onMarkAllAsUnread = () => {
+    dispatch(markAllNotificationsAsUnread());
+    setIsDropdownOpen(false);
+  };
+
+  const onFilterSelect = (chosenFilter: string) => {
+    activeFilters.includes(chosenFilter)
+      ? setActiveFilters(activeFilters.filter((filter) => filter !== chosenFilter))
+      : setActiveFilters([...activeFilters, chosenFilter]);
+  };
+
+  const dropdownItems = [
+    <DropdownItem key="read all" onClick={() => onMarkAllAsRead()}>
+      Mark visible as read
+    </DropdownItem>,
+    <DropdownItem key="unread all" onClick={() => onMarkAllAsUnread()}>
+      Mark visible as unread
+    </DropdownItem>,
+    <Divider key="divider" />,
+    <DropdownItem key="event log">
+      <Icon>
+        <ExternalLinkSquareAltIcon />
+      </Icon>
+      View event log
```

Comment:
Based on the mocks, this should be fliped
```suggestion
      View event log
      <Icon>
        <ExternalLinkSquareAltIcon />
      </Icon>
```